packages/neo-one-client-core/src/user/DapiUserAccountProvider.ts

```diff
@@ -0,0 +1,577 @@
+    await this.initPromise;
+
+    return this.provider.getBlockCount(network);
+  }
+
+  public async getAccount(network: NetworkType, address: AddressString): Promise<Account> {
+    await this.initPromise;
+    const balances = await this.dapi.getBalance({ params: { address, assets: [Hash256.NEO, Hash256.GAS] }, network });
+
+    return {
+      address,
+      balances: balances[address].reduce<{ readonly [asset: string]: BigNumber }>(
+        (acc, balance) => ({
+          ...acc,
+          [balance.assetID]: new BigNumber(balance.amount),
+        }),
+        {},
+      ),
+    };
+  }
+
+  public iterBlocks(network: NetworkType, options?: IterOptions): AsyncIterable<Block> {
+    if (!this.mutableInitialized) {
+      throw new NotIntializedError();
+    }
+
+    return this.provider.iterBlocks(network, options);
+  }
+
+  public iterActionsRaw(network: NetworkType, options?: IterOptions): AsyncIterable<RawAction> {
+    if (!this.mutableInitialized) {
+      throw new NotIntializedError();
+    }
+
+    if (this.provider.iterActionsRaw === undefined) {
+      throw new NotImplementedError('iterActionsRaw');
+    }
+
+    return this.provider.iterActionsRaw(network, options);
+  }
+
+  public async transfer(
+    transfers: ReadonlyArray<Transfer>,
+    options?: TransactionOptions,
+  ): Promise<TransactionResult<TransactionReceipt, InvocationTransaction>> {
+    await this.initPromise;
+    if (transfers.length !== 1) {
+      throw new NotImplementedError('Multiple transfers per transaction');
+    }
+    const { from, networkFee, systemFee } = this.getTransactionOptions(options);
+    const network = this.defaultNetworkInternal$.getValue() as NetworkType;
+
+    const { amount, asset, to } = transfers[0];
+
+    const { txid } = await this.dapi.send({
+      fromAddress: from.address,
+      toAddress: to,
+      asset,
+      amount: amount.toString(),
+      fee: systemFee.plus(networkFee).toString(),
+      network,
+    });
+    const transaction = (await this.provider.getTransaction(network, txid)) as InvocationTransaction;
+
+    return {
+      transaction,
+      confirmed: async () => this.provider.getTransactionReceipt(network, txid),
+    };
+  }
+
+  public async claim(_options?: TransactionOptions): Promise<TransactionResult<TransactionReceipt, ClaimTransaction>> {
+    throw new NotImplementedError('claim');
+  }
+
+  public async invoke(
+    contract: AddressString,
+    method: string,
+    params: ReadonlyArray<ScriptBuilderParam | undefined>,
+    _paramsZipped: ReadonlyArray<readonly [string, Param | undefined]>,
+    verify: boolean,
+    options: InvokeSendUnsafeReceiveTransactionOptions = {},
+    _sourceMaps?: Promise<SourceMaps>,
+  ): Promise<TransactionResult<RawInvokeReceipt, InvocationTransaction>> {
+    await this.initPromise;
+    const { from, networkFee, systemFee } = this.getTransactionOptions(options);
+    const network = this.defaultNetworkInternal$.getValue() as NetworkType;
```

Comment:
Why is this using `defaultNetworkInternal` instead of `from.network`?

I think this could be refactored a bit - all of these invoke methods should be identical (and in fact, almost are identical) between this class and `LocalUserAccountProvider` - the only difference is `invokeRaw` + the `await initPromise` at the top. I think you should be able to move all of the `invoke` methods into the base class with an abstract `invokeRaw` method and just call `await initPromise` in the implementation for this class.